splink/analyse_blocking.py

```diff
@@ -57,3 +72,86 @@ def number_of_comparisons_generated_by_blocking_rule_sql(
     """
 
     return sql
+
+
+def cumulative_comparisons_generated_by_blocking_rules(
+    linker: "Linker",
+    concat_df,
+    blocking_rules,
+    link_type=None,
+    unique_id_column_name=None,
+):
+
+    where_condition = generate_blocking_rule_where_condition(
+        linker,
+        link_type,
+        unique_id_column_name,
+    )
+
+    if isinstance(blocking_rules[0], BlockingRule):
+        brs_as_objs = blocking_rules
+    else:
+        brs_as_objs = []
+        for br in blocking_rules:
+            br = BlockingRule(br)
+            br.preceding_rules = brs_as_objs.copy()
+            brs_as_objs.append(br)
+
+    # Calculate the Cartesian Product
+    sql = f"""
+        select count(*) as count
+        from {concat_df}
+    """
+    df_count = linker._enqueue_and_execute_sql_pipeline(
+        sql, "__splink__analyse_blocking_rule"
+    )
+    row_count_df = df_count.as_record_dict()[0]
+    df_count.drop_table_from_database()
+
+    cartesian = calculate_cartesian(row_count_df["count"])
+
+    br_comparisons = []
+    cumulative_sum = 0
+
+    # return sqls
+    for br in brs_as_objs:
```

Comment:
Would it be possible to re-use code like in `block_using_rules_sql` (used e.g. in `linker.deterministic_link`), and then add a final step to the pipeline where you do a groupby on the match key?